app/uploaders/image_four_uploader.rb

```diff
@@ -0,0 +1,27 @@
+# encoding: utf-8
+
+class ImageFourUploader < CarrierWave::Uploader::Base
+  version :default
+
+  if Rails.env.production? || Rails.env.development?
+    storage :fog
+  else
+    storage :file
+  end
+
+  def store_dir
+    "uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
+  end
+
+  def extension_white_list
+    %w(jpg jpeg gif png)
+  end
+
+  include CarrierWave::MiniMagick
+
+  process :resize_to_fit => [500, 500]
+
+  version :thumb do
+    process :resize_to_fill => [200, 200]
+  end
+end
```

Comment:
Inheritance allows us to reuse methods from another class. For instance, when one of our models inherits from ActiveRecord, we gain access to the `.all` and `.where` methods.

I noticed that all of your image uploader classes looked very similar. Instead of duplicating code, you can create one `image_uploader.rb` class and then inherit from it.

```
class ImageFourUploader << ImageUploader
end
```